### PR TITLE
redirect to default app after solving the 2FA challenge

### LINKED ISSUE #52: Redirect after login with TOTP
It would be fantastic if TOTP checks if an alternate app is configured in ownCloud/nextcloud config als "default" app and redirects to it instead of an hardcoded redirect to the files app.

Thanks for your great work!

Ralf

### PR DESCRIPTION
**Pull request by ChristophWurst**
fixes https://github.com/ChristophWurst/twofactor_totp/issues/52

### CODE REVIEW

File: tests/Core/Controller/TwoFactorChallengeControllerTest.php [hunk 1]
```diff
@@ -167,13 +167,8 @@ public function testSolveChallenge() {
...
 			->with('myprovider', $user, 'token')
 			->will($this->returnValue(true));
-		$this->urlGenerator->expects($this->once())
-			->method('linkToRoute')
-			->with('files.view.index')
-			->will($this->returnValue('files/index/url'));
 
-		$expected = new \OCP\AppFramework\Http\RedirectResponse('files/index/url');
-		$this->assertEquals($expected, $this->controller->solveChallenge('myprovider', 'token'));
+		$this->assertInstanceOf('\OCP\AppFramework\Http\RedirectResponse', $this->controller->solveChallenge('myprovider', 'token'));
```
Comment (ChristophWurst) [inline tests/Core/Controller/TwoFactorChallengeControllerTest.php]: is there a way to reliably determine the output of ``OC_Util::getDefaultPageUrl()`` while testing? Then I could compare the actual return values here.

Comment (LukasReschke) [inline tests/Core/Controller/TwoFactorChallengeControllerTest.php]: I don't think so, but you can just make $expected use `OC_Util::getDefaultPageUrl()` as well. Not perfect, but works ;)

**Comment by rullzer** [comment]
LGTM

**Comment by LukasReschke** [comment]
LGTM

**Comment by juliushaertl** [comment]
This is a bug fix, so i guess we should backport this? cc @karlitschek @ChristophWurst 

Bug on stable9 reported on the forums: https://help.nextcloud.com/t/apporder-and-totp-wrong-landing-page/3252

**Comment by nickvergessen** [comment]
stable9 doesnt have 2 factor, therefor the fix is different.
But was this backported to 10? @ChristophWurst

**Comment by ChristophWurst** [comment]
I don't think this was backported, or at least I didn't do it

**Comment by juliushaertl** [comment]
@nickvergessen Ah of course, i mean stable10

**Comment by xraMsamohT** [comment]
@ChristophWurst @nickvergessen Sorry, this was my fault. I posted a wrong version number in the forum. The correct version is 9.1.0.16.

**Comment by karlitschek** [comment]
I think a backport makes sense! 👍

**Pull request merged**